### PR TITLE
fix(core): fix rollback failure

### PR DESCRIPTION
**Pull request by ideoma**
Fix rare SIGBUS which can happen after a transaction rollback on Linux.

When file is truncated it can be pad with 0 to the page size but if the file is already truncated via another file descriptor SIGBUS crash can happen on Linux.

To avoid, close active partition on rollbacks and also check file size before padding with 0 on truncate.

### CODE REVIEW

File: core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java [hunk 1]
```diff
@@ -73,7 +73,9 @@ public void close(boolean truncate, byte truncateMode) {
             if (truncate) {
                 long appendOffset = getAppendOffset();
                 truncateSize = truncateMode == Vm.TRUNCATE_TO_PAGE ? Files.ceilPageSize(appendOffset) : appendOffset;
-                long sz = Math.min(size, truncateSize);
+                // If this is lazy close of unused memory the underlying file can be already truncated using another fd
+                // and memset can lead to SIGBUS on Linux. Check the phisical file length before trying to memset to the map
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java:77]:
```suggestion
                // and memset can lead to SIGBUS on Linux. Check the physical file length before trying to memset to the map
```

File: core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java [hunk 2]
```diff
@@ -73,7 +73,9 @@ public void close(boolean truncate, byte truncateMode) {
...
+                long sz = Math.min(ff.length(fd), Math.min(size, truncateSize));
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java:78]: Unlike other ff calls used here, `ff.length()` may actually throw. We may want to handle the exception here (or introduce a non-throwing version of `length()`) and log it.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCMARWImpl.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java]: done

File: core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java (hunk 2 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java]: done, added a catch here

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCMARWImpl.java|9|9|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCMARWImpl.java|9|9|100.00%|

**Review by puzpuzpuz — APPROVED** [review]

**Pull request merged**